### PR TITLE
[AAE-1137] Refactor delete process directive

### PR DESCRIPTION
**Pull request by arditdomi**
**Please check if the PR fulfills these requirements**

> - [X] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [X] Tests for the changes have been added (for bug fixes / features)
> - [X] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [X] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/AAE-1137

**What is the new behaviour?**
- The delete process directive requires additional changes apart from the initial implementation that this ticket describes. It should not receive the process details as @Inputs(). It should automatically get the process details every time that the getProcessDefinitionById method is called by subscribing to the Observable that notifies that the process details have changed every time that the method is called.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/process-services-cloud/src/lib/process/directives/cancel-process.directive.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            this.error.emit(error);
+        }
+    }
+
+    private async cancelProcess() {
+        const currentUser: string = this.identityUserService.getCurrentUserInfo().username;
+        if (this.processCloudService.canCancelProcess(this.processInstanceDetails, currentUser)) {
+            await this.processCloudService.cancelProcess(this.processInstanceDetails.appName, this.processInstanceDetails.id)
+                .subscribe((response) => {
+                    this.success.emit(response);
```
Comment (davidcanonieto) [inline lib/process-services-cloud/src/lib/process/directives/cancel-process.directive.ts]: Can we also unsubscribe from this observable? `.pipe(takeUntil(this.onDestroy$))`

File: lib/process-services-cloud/src/lib/process/services/process-cloud.service.ts [hunk 2]
```diff
@@ -17,34 +17,72 @@
...
+import { ProcessInstanceCloud } from '../start-process/models/process-instance-cloud.model';
 import { BaseCloudService } from '../../services/base-cloud.service';
 
 @Injectable({
     providedIn: 'root'
 })
 export class ProcessCloudService extends BaseCloudService {
 
+    private dataChangesDetected = new Subject();
+    dataChangesDetected$: Observable<object>;
```
Comment (mauriziovitale) [inline lib/process-services-cloud/src/lib/process/services/process-cloud.service.ts]: object? are you a JAVA developer ?

File: lib/process-services-cloud/src/lib/process/services/process-cloud.service.ts [hunk 3]
```diff
@@ -17,34 +17,72 @@
...
+import { map } from 'rxjs/operators';
+import { ProcessInstanceCloud } from '../start-process/models/process-instance-cloud.model';
 import { BaseCloudService } from '../../services/base-cloud.service';
 
 @Injectable({
     providedIn: 'root'
 })
 export class ProcessCloudService extends BaseCloudService {
 
+    private dataChangesDetected = new Subject();
```
Comment (mauriziovitale) [inline lib/process-services-cloud/src/lib/process/services/process-cloud.service.ts]: can you please use BehaviourSubject?

File: docs/process-services-cloud/directives/cancel-process.directive.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+---
+
+# [Cancel process directive](../../../lib/process-services-cloud/src/lib/process/directives/cancel-process.directive.ts "Defined in cancel-process.directive.ts")
+
+Cancels a process
+
+## Basic Usage
+
+```html
+<button adf-cloud-delete-process (success)="onProcessCancelled()" (error)="onCancelProcessError()" (canCancelProcess)="showCancelAction()">Delete</button>
```
Comment (mauriziovitale) [inline docs/process-services-cloud/directives/cancel-process.directive.md]: please use the correct directive name

File: lib/process-services-cloud/src/lib/process/directives/cancel-process.directive.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    constructor(
+        private processCloudService: ProcessCloudService,
+        private identityUserService: IdentityUserService) {}
+
+    ngOnInit() {
+        this.processCloudService.dataChangesDetected$
+            .pipe(takeUntil(this.onDestroy$))
+            .subscribe((processDetails: ProcessInstanceCloud) => {
+                this.processInstanceDetails = processDetails;
+                const currentUser: string = this.identityUserService.getCurrentUserInfo().username;
```
Comment (mauriziovitale) [inline lib/process-services-cloud/src/lib/process/directives/cancel-process.directive.ts]: can you call it just once ?

File: lib/process-services-cloud/src/lib/process/directives/cancel-process.directive.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    @Output()
+    success: EventEmitter<any> = new EventEmitter<any>();
+
+    /** Emitted when the process cannot be cancelled. */
+    @Output()
+    error: EventEmitter<any> = new EventEmitter<any>();
+
+    @Output()
+    /** Emitted when the can cancel process status is checked */
+    canCancelProcess: EventEmitter<boolean> = new EventEmitter<boolean>();
```
Comment (mauriziovitale) [inline lib/process-services-cloud/src/lib/process/directives/cancel-process.directive.ts]: not sure we need this event. The main goal of this directive is to change the beheviour of the html and disable/enable the button based on the canCancel.
I would suggest using something similar
`    this.renderer.setAttribute(this.el.nativeElement, 'disabled', 'true');`

**Pull request merged**